R/multiple-conditions.R

```diff
@@ -0,0 +1,182 @@
+#' @title Multiple chronic conditions
+#' 
+#' @description This function generates a derived variable (multiple_conditions)
+#'  that counts the number of chronic conditions a respondent has. This function
+#'  takes 5 CCHS variables (heart disease, cancer, stroke, bowel disorder, and
+#'  arthritis), and well one derived variable (respiratory condition) to count
+#'  the number of conditions a respondent has.
+#'  
+#' @param CCC_121 variable indicating if respondent has heart disease
+#' 
+#' @param CCC_131 variable indicating if respondent has active cancer
+#' 
+#' @param CCC_151 variable indicating if respondent suffers from the effects
+#'  of a stroke
+#' 
+#' @param CCC_171 variable indicating if respondent has a bowel disorder
+#' 
+#' @param resp_condition_der derived variable indicating if respondent has a
+#'  respiratory condition. See \code{\link{resp_condition_fun1}} for
+#'  documentation on how variable was derived
+#'
+#' @param CCC_051 variable indicating if respondent has arthritis or
+#'  rheumatism
+#'
+#' @details mood disorder (CCC_280) was not asked to respondents in the 2001
+#'  CCHS survey cycle. This mean respondents in this cycle will only be able to
+#'  have a maximum of 6 chronic conditions as opposed to 7 for respondents in
+#'  other cycles. \code{\link{multiple_conditions_fun2}} is used for CCHS cycles
+#'  from 2003 to 2014.
+#' 
+#' @return A categorical variable indicating the number of chronic conditions
+#'  a respondent has. Respondents with 5 or more conditions are grouped in the
+#'  "5+" category. 
+#' 
+#' @examples 
+#'  # Using rec_with_table() to generate multiple_conditions in a CCHS
+#'  # cycle.
+#'  
+#'  # multiple_conditions_fun1() is specified in variable_details.csv along with
+#'  # the CCHS variables and cycles included.
+#'  
+#'  # To generate multiple_conditions, use rec_with_table() and specify the
+#'  # multiple_conditions, along with the variables that are derived from it.
+#'  # Since resp_condition_der is also a derived variable, you will have to
+#'  # specify the variables that are derived from it. In this example, data
+#'  # from the 2001 CCHS will be used, so DHHGAGE_cont, CCC_091, and CCC_91A,
+#'  # and CCC_031 will be specified along with resp_condition_der.
+#'  
+#' library(cchsflow)
+#'  conditions_2001 <- rec_with_table(cchs2001_p, c("DHHGAGE_cont", "CCC_091",
+#'  "CCC_91A", "CCC_031", "CCC_121","CCC_131","CCC_151", "CCC_171","CCC_280",
+#'  "resp_condition_der","CCC_051", "multiple_conditions"))
+#'  
+#'  head(conditions_2001)
+#' 
+#' @seealso \code{\link{multiple_conditions_fun2}}     
+#' @export
+
+multiple_conditions_fun1 <- 
+  function(CCC_121,CCC_131,CCC_151,CCC_171,resp_condition_der,CCC_051){
+    
+    # Convert variables to numeric
+    CCC_121 <- as.numeric(CCC_121)
```

Comment:
What will happen if the CCC variables are missing? Will this fail gracefully?

Do we need checks, with warnings and errors? Is there a testthat?

Reply:
This would be the error produced if a CCC variable was missing
`Error in multiple_conditions_fun2(CCC_121 = 1, CCC_131 = 2, CCC_151 = 2,  : 
  argument "CCC_280" is missing, with no default`
This is why I'm in favour of not adding defaults as the error message outputted signals a user that they're missing a variable.

Testthat file for the two functions  can be found [here](https://github.com/Big-Life-Lab/cchsflow/blob/dev/tests/testthat/test-multiple-conditions.R)

Reply:
That is a nice error and would suffice. Does R stop with that error or continue to caluculate then next variable transformation?

Does adding a defult break that error message?

Reply:
It does stop the process but since the function call is handled by rec_with_table that handles the fail by telling u which column is missing and that variable is missing and it never makes it to the function. If the requested function variables are present they are fed to the function only issues occur when variable_start is not same order as the arguments here. And this being mostly an internal function we have no ways of really setting default values(assuming u want it to pull value from database) if u want it to be an abosulute default value that can be done but i dont see the purpose.

Reply:
To add what @DarkShadeKnigh said, adding defaults will remove that error message.